### PR TITLE
WIP - Update pipeline group entity #2026 

### PR DESCRIPTION
**Pull request by iamsumanth**
Work in progress.
After updating the authorization, validation needs to be done. I will do it as a part of next pull request.

### CODE REVIEW

File: server/src/com/thoughtworks/go/config/update/PipelineConfigsUpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.thoughtworks.go.config.CruiseConfig;
+import com.thoughtworks.go.config.PipelineConfigs;
+import com.thoughtworks.go.config.UpdateConfigCommand;
+import com.thoughtworks.go.config.commands.EntityConfigUpdateCommand;
+import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.GoConfigService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.validation.ConfigUpdateValidator;
+
+public class PipelineConfigsUpdateCommand implements EntityConfigUpdateCommand<PipelineConfigs> {
```
Comment (varshavaradarajan) [inline server/src/com/thoughtworks/go/config/update/PipelineConfigsUpdateCommand.java]: Rename to UpdatePipelineGroupsConfigCommand

File: server/test/integration/com/thoughtworks/go/server/service/GoConfigServiceIntegrationTest.java [hunk 2]
```diff
@@ -788,6 +791,38 @@ public void shouldNotThrowUpOnConfigSave_ViaNormalFlow() throws Exception {
...
+    public void shouldReturnFalseIfPipelineGroupIsAvailableAndUserIsNotAdminOfGroup() throws IOException {
+        setupSecurity();
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        boolean canEditPipelineGroup = goConfigService.canEditPipelineGroup("defaultGroup", new Username(new CaseInsensitiveString("loser")), result);
+
+        assertFalse(canEditPipelineGroup);
+        assertThat(result.isSuccessful(), is(false));
+    }
+
+
```
Comment (varshavaradarajan) [inline server/test/integration/com/thoughtworks/go/server/service/GoConfigServiceIntegrationTest.java]: shouldn't these be in GoConfigServiceTest?

File: server/src/com/thoughtworks/go/server/service/GoConfigService.java [hunk 3]
```diff
@@ -1132,4 +1132,12 @@ public ConfigElementImplementationRegistry getRegistry() {
         return registry;
     }
 
+    public boolean canEditPipelineGroup(String groupName, Username username, HttpLocalizedOperationResult result) {
+        CruiseConfig cruiseConfig = getCurrentConfig();
+        if((isValidGroup(groupName, cruiseConfig, result) && isAdminOfGroup(groupName, username, result)) == true) {
+            return true;
+        }
+        return false;
+    }
```
Comment (GaneshSPatil) [inline server/src/com/thoughtworks/go/server/service/GoConfigService.java]: can't we do `return (isValidGroup(groupName, cruiseConfig, result) && isAdminOfGroup(groupName, username, result))`?

File: server/src/com/thoughtworks/go/server/service/PipelineConfigsService.java [hunk 4]
```diff
@@ -113,4 +116,17 @@ private boolean userHasPermissions(Username username, String groupName, HttpLoca
...
+    public void updateAuthorization(Authorization authorization, String groupName, HttpLocalizedOperationResult result, Username validUser) {
+        UpdateAuthorizationCommand command = new UpdateAuthorizationCommand(groupName, authorization);
+        try{
+            PipelineConfigsUpdateCommand updateCommand = new PipelineConfigsUpdateCommand(goConfigService, command, result, validUser, groupName);
+            goConfigService.updateConfig(updateCommand, validUser);
+        } catch (Exception e) {
+            if (e instanceof GoConfigInvalidException) {
+                result.unprocessableEntity(LocalizedMessage.string("ENTITY_CONFIG_VALIDATION_FAILED"));
+            } else {
+                result.internalServerError(LocalizedMessage.string("INTERNAL_SERVER_ERROR"));
```
Comment (GaneshSPatil) [inline server/src/com/thoughtworks/go/server/service/PipelineConfigsService.java]: can we also show the error along with the message
`result.internalServerError(LocalizedMessage.string("INTERNAL_SERVER_ERROR", e.getMessage()));`

**Comment by ketan** [comment]
What's the status of this? Is this needed?

**Comment by ketan** [comment]
> What's the status of this? Is this needed?

@iamsumanth?

**Comment by iamsumanth** [comment]
@ketan, we can close this pull request. I was working on this branch long back but based on the priority I didn't look at it later. We can close this for now whenever we need this feature we can come back again.

**Pull request closed without merging**